std/utf.d

```diff
@@ -1005,8 +1005,7 @@ alias UseReplacementDchar = Flag!"useReplacementDchar";
         sequence and useReplacementDchar is UseReplacementDchar.no
   +/
 dchar decode(UseReplacementDchar useReplacementDchar = UseReplacementDchar.no, S)(auto ref S str, ref size_t index)
-    if (!isSomeString!S &&
-        isRandomAccessRange!S && hasSlicing!S && hasLength!S && isSomeChar!(ElementType!S))
+    if (isSomeString!S || isRandomAccessRange!S && hasSlicing!S && hasLength!S && isSomeChar!(ElementType!S))
```

Comment:
The other template constraint explicitly excludes strings - now they are allowed. Is this on purpose?